src/components/Context.js

```diff
@@ -206,16 +207,58 @@ class ProductProvider extends Component {
           }
         });
       });
-
       this.dispatch(SET_COUPON, {
         appliedVoucher: voucher,
       });
       toast.success("Promotion applied");
+      console.log(this.state.cart);
     } catch (e) {
       toast.error("Promotion not found");
     }
   };
 
+  checkoutCart = async () => {
+    try {
+      if (!_.isEmpty(this.state.appliedVoucher)) {
+        const prepareItemsPayload = (item) => {
+          return {
+            product_id: item.id,
+            quantity: item.count,
+            price: item.price,
+            amount: item.total,
+          };
+        };
+        const redemptionPayload = {
+          customer: {
+            tracking_id: this.state.appliedVoucher.tracking_id,
+          },
+          order: {
+            amount: this.state.cartTotalAfterPromotion,
+            items: this.state.cart.map(prepareItemsPayload),
+          },
+        };
+
+        const code = this.state.appliedVoucher.code;
+
+        await new Promise((resolve, reject) => {
+          window.Voucherify.redeem(code, redemptionPayload, (response) => {
+            console.log(response);
+            if (response.result === "SUCCESS") {
+              this.clearCart();
```

Comment:
`clearCart()` displays a toast "Cart cleared" which is not the message you want to see after a successful purchase.

let's not abuse `clearCart()` for more cases than it was designed for :)

i guess we could `dispatch(CLEAR_CART)` and `toast.info(payment successful)`

Reply:
PS I'd make it this way:

```
if response == SUCCESS {
  resolve(response)
} else {
  reject(...)
}
```

and then

```
const redemption = await new Promise(...) // well you don't need the redemption, so we can skip it
this.clearCart() // or dispatch...
```

I'm struggling to explain it properly, but this way we keep any unnecessary logic away from the Promise body. Now that piece of our logic is somehow hidden, wrapped into that Promise.

Reply:
but overall it's a very minor thing, merely a suggestion.